src/System.Private.CoreLib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -240,12 +247,12 @@ public bool Trim()
                 // Under high pressure, release all thread locals
                 if (log.IsEnabled())
                 {
-                    foreach (KeyValuePair<T[][], object> tlsBuckets in s_allTlsBuckets)
+                    foreach (KeyValuePair<T[][], object?> tlsBuckets in s_allTlsBuckets)
```

Comment:
Why is this not `KeyValuePair<T[]?[], object?>`

Reply:
since `ConditionalWeakTable<T[]?[], object?>? s_allTlsBuckets` and also below `T[]?[] buckets = tlsBuckets.Key;`

Reply:
same below.

Reply:
Fixed.  Not sure why the compiler didn't warn about this, either.